src/services/signatureHelp.ts

```diff
@@ -205,431 +205,431 @@ namespace ts.SignatureHelp {
+                return createJavaScriptSignatureHelpItems(argumentInfo, program);
             }
 
             return undefined;
         }
 
-        return createSignatureHelpItems(candidates, resolvedSignature, argumentInfo);
+        return createSignatureHelpItems(candidates, resolvedSignature, argumentInfo, typeChecker);
+    }
 
-        function createJavaScriptSignatureHelpItems(argumentInfo: ArgumentListInfo): SignatureHelpItems {
-            if (argumentInfo.invocation.kind !== SyntaxKind.CallExpression) {
-                return undefined;
-            }
+    function createJavaScriptSignatureHelpItems(argumentInfo: ArgumentListInfo, program: Program): SignatureHelpItems {
+        if (argumentInfo.invocation.kind !== SyntaxKind.CallExpression) {
+            return undefined;
+        }
 
-            // See if we can find some symbol with the call expression name that has call signatures.
-            let callExpression = <CallExpression>argumentInfo.invocation;
-            let expression = callExpression.expression;
-            let name = expression.kind === SyntaxKind.Identifier
-                ? <Identifier> expression
-                : expression.kind === SyntaxKind.PropertyAccessExpression
-                    ? (<PropertyAccessExpression>expression).name
-                    : undefined;
+        // See if we can find some symbol with the call expression name that has call signatures.
+        let callExpression = <CallExpression>argumentInfo.invocation;
+        let expression = callExpression.expression;
+        let name = expression.kind === SyntaxKind.Identifier
+            ? <Identifier>expression
+            : expression.kind === SyntaxKind.PropertyAccessExpression
+                ? (<PropertyAccessExpression>expression).name
+                : undefined;
 
-            if (!name || !name.text) {
-                return undefined;
-            }
+        if (!name || !name.text) {
+            return undefined;
+        }
 
-            let typeChecker = program.getTypeChecker();
-            for (let sourceFile of program.getSourceFiles()) {
-                let nameToDeclarations = sourceFile.getNamedDeclarations();
-                let declarations = getProperty(nameToDeclarations, name.text);
-
-                if (declarations) {
-                    for (let declaration of declarations) {
-                        let symbol = declaration.symbol;
-                        if (symbol) {
-                            let type = typeChecker.getTypeOfSymbolAtLocation(symbol, declaration);
-                            if (type) {
-                                let callSignatures = type.getCallSignatures();
-                                if (callSignatures && callSignatures.length) {
-                                    return createSignatureHelpItems(callSignatures, callSignatures[0], argumentInfo);
-                                }
+        let typeChecker = program.getTypeChecker();

```

Comment:
I think this should be `const`. We would want to add this files into linting maybe